File: nhglib-core/src/io/github/movementspeed/nhglib/utils/data/Bundle.java
package io.github.movementspeed.nhglib.utils.data;

import java.util.HashMap;

/**
 * Created by Fausto Napoli on 25/10/2016.
 */
public class Bundle extends HashMap<String, Object> {
    public boolean getBoolean(String key) {
        return getBoolean(key, false);
    }

    public boolean getBoolean(String key, boolean defaultValue) {
        boolean res = defaultValue;
        Object value = get(key);

        if (value instanceof Boolean) {
            res = (boolean) value;
        }

        return res;
    }

    public byte getByte(String key) {
        return getByte(key, (byte) -1);
    }

    public byte getByte(String key, byte defaultValue) {
        byte res = defaultValue;
        Object value = get(key);

        if (value instanceof Byte) {
            res = (byte) value;
        }

        return res;
    }

    public short getShort(String key) {
        return getShort(key, (short) -1);
    }

    public short getShort(String key, short defaultValue) {
        short res = defaultValue;
        Object value = get(key);

        if (value instanceof Short) {
            res = (short) value;
        }

        return res;
    }

    public int getInteger(String key) {
        return getInteger(key, -1);
    }

    public int getInteger(String key, int defaultValue) {
        int res = defaultValue;
        Object value = get(key);

        if (value instanceof Integer) {
            res = (int) value;
        }

        return res;
    }

    public long getLong(String key) {
        return getLong(key, -1L);
    }

    public long getLong(String key, long defaultValue) {
        long res = defaultValue;
        Object value = get(key);

        if (value instanceof Long) {
            res = (long) value;
        }

        return res;
    }

    public float getFloat(String key) {
        return getFloat(key, -1f);
    }

    public float getFloat(String key, float defaultValue) {
        float res = defaultValue;
        Object value = get(key);

        if (value instanceof Float) {
            res = (float) value;
        }

        return res;
    }

    public double getDouble(String key) {
        return getDouble(key, -1D);
    }

    public double getDouble(String key, double defaultValue) {
        double res = defaultValue;
        Object value = get(key);

        if (value instanceof Double) {
            res = (double) value;
        }

        return res;
    }

    public char getCharacter(String key) {
        return getCharacter(key, ' ');
    }

    public char getCharacter(String key, char defaultValue) {
        char res = defaultValue;
        Object value = get(key);

        if (value instanceof Character) {
            res = (char) value;
        }

        return res;
    }

    public String getString(String key) {
        return getString(key, null);
    }

    public String getString(String key, String defaultValue) {
        String res = null;
        Object value = get(key);

        if (value instanceof String) {
            res = (String) value;
        }

        if (res == null) {
            res = defaultValue;
        }

        return res;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/utils/data/Bundle.kt
package io.github.movementspeed.nhglib.utils.data

/**
 * Created by Fausto Napoli on 25/10/2016.
 */
class Bundle : HashMap<String, Any>() {

    @JvmOverloads
    fun getBoolean(key: String, defaultValue: Boolean = false): Boolean {
        var res = defaultValue
        val value = get(key)

        if (value is Boolean) {
            res = value
        }

        return res
    }

    @JvmOverloads
    fun getByte(key: String, defaultValue: Byte = (-1).toByte()): Byte {
        var res = defaultValue
        val value = get(key)

        if (value is Byte) {
            res = value
        }

        return res
    }

    @JvmOverloads
    fun getShort(key: String, defaultValue: Short = (-1).toShort()): Short {
        var res = defaultValue
        val value = get(key)

        if (value is Short) {
            res = value
        }

        return res
    }

    @JvmOverloads
    fun getInteger(key: String, defaultValue: Int = -1): Int {
        var res = defaultValue
        val value = get(key)

        if (value is Int) {
            res = value
        }

        return res
    }

    @JvmOverloads
    fun getLong(key: String, defaultValue: Long = -1L): Long {
        var res = defaultValue
        val value = get(key)

        if (value is Long) {
            res = value
        }

        return res
    }

    @JvmOverloads
    fun getFloat(key: String, defaultValue: Float = -1f): Float {
        var res = defaultValue
        val value = get(key)

        if (value is Float) {
            res = value
        }

        return res
    }

    @JvmOverloads
    fun getDouble(key: String, defaultValue: Double = -1.0): Double {
        var res = defaultValue
        val value = get(key)

        if (value is Double) {
            res = value
        }

        return res
    }

    @JvmOverloads
    fun getCharacter(key: String, defaultValue: Char = ' '): Char {
        var res = defaultValue
        val value = get(key)

        if (value is Char) {
            res = value
        }

        return res
    }

    @JvmOverloads
    fun getString(key: String, defaultValue: String? = null): String? {
        var res: String? = null
        val value = get(key)

        if (value is String) {
            res = value
        }

        if (res == null) {
            res = defaultValue
        }

        return res
    }
}
